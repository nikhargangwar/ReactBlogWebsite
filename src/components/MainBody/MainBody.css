.main {

    flex: 1;
    /* //height: 79vh; */

}



.cardArea {
    width: 70%;
    margin: 0 auto;
    display: flex;
    flex-wrap: wrap;
    min-height: 79vh;
    padding-top: 15px;
    justify-content: flex-start;
    justify-content: space-around;
    /* gap: 0px 40px; */
    gap: 0px 50px;
}
