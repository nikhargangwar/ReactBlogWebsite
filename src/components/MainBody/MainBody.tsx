/* eslint-disable react/jsx-key */
import './MainBody.css';
import { useState } from 'react';
import Card from '../Card/Card' // Import
import {stockData} from '../../assets/data';
// import * as cardImage from '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/images/';
// const imgAdd = '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/images/';

function MainBody() {

  return (


    <div className="cardArea">
     {
      stockData.map((task)=>

        <Card 
          date={task.date}
          readingTime={task.readingTime}
          title = {task.title}
          description={task.description}
          claps={task.claps}
          liked={task.liked}
          image={task.image}
        />
      )
     }
      
    </div>
  );
}

export default MainBody;
