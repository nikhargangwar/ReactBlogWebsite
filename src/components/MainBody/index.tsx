/* eslint-disable react/jsx-key */
import './MainBody.css';
import {useContext, useEffect, useState } from 'react';
import makeRequest from '../../utils/makeRequest';
import { GET_BLOG_DATA } from '../../constants/apiEndPoints';
import Card from '../Card' // Import
import { clearConfigCache } from 'prettier';
import { isTypeAliasDeclaration } from 'typescript';
import { useNavigate } from 'react-router-dom';
import { BlogPostContext } from '../../context/BlogPostContext';

// import {stockData} from '../../assets/data';
// import * as cardImage from '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/images/';
// const imgAdd = '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/images/';

interface BlogData {
  id: number;
  image: string;
  date: string;
  readingTime: string;
  title: string;
  description: string;
  claps: number;
  liked: boolean;
}

function MainBody() {
  // const [blogData, setBlogData] = useState<BlogData[]>();
  const {allBlogData,setAllBlogData} = useContext(BlogPostContext)
  const [error, setError] = useState();
  const navigate = useNavigate()

    useEffect(() => {
     
      makeRequest(GET_BLOG_DATA,{},navigate)
        .then((response) => {
          setAllBlogData(response);
        })
        .catch((e) => {
          setError(e.message);
        });
    }, []);

  if (error) {
    return (
      <div className="blogDataError" data-testid="error">
        <p>{error}</p>
      </div>
    );
  }


  return allBlogData ? (

    <div className="cardArea">
     {
      allBlogData.map((task)=>

        <Card key={task.id}
          id={task.id}
          date={task.date}
          readingTime={task.readingTime}
          title = {task.title}
          description={task.description}
          claps={task.claps}
          liked={task.liked}
          image={task.image}
        />
      )
     }
      
    </div>
  ): (
    <div className="blogDataLoader">
      <p>Loading...</p>
    </div>
  );;
}


export default MainBody;