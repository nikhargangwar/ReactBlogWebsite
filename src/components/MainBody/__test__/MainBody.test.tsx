import React from 'react';
import { render } from '@testing-library/react';

import MainBody from '..';

describe('MainBody', () => {
  // 1
  it('should render when component is mounted', () => {
    const { container } = render(<MainBody />);
    expect(container).toMatchSnapshot();
  });
});
