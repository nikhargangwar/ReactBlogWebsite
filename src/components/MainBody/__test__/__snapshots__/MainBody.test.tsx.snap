// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainBody should render when component is mounted 1`] = `
<div>
  <div
    class="cardArea"
  >
    <div
      class="cardMain"
    >
      <div
        class="cardImage"
      >
        <img
          src="abstract.png"
        />
      </div>
      <div
        class="cardTime"
      >
        <h5>
          2nd Januray, 2018
        </h5>
        <h5>
          2 mins
        </h5>
      </div>
      <div
        class="cardTitle"
      >
        <h3>
          The future of abstract art and the culture ...
        </h3>
      </div>
      <div
        class="cardDescription"
      >
        <h5>
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </h5>
      </div>
      <hr
        id="cardLine"
      />
      <div
        class="cardReaction"
      >
        <div
          class="clapCount"
        >
          <img
            class="clap-button"
            src="clapping.svg"
          />
          <h5
            class="clap"
          >
            10
          </h5>
        </div>
        <img
          alt="blackHeart"
          class="like"
          src="heart-black.svg"
        />
      </div>
    </div>
    <div
      class="cardMain"
    >
      <div
        class="cardImage"
      >
        <img
          src="outdoor.png"
        />
      </div>
      <div
        class="cardTime"
      >
        <h5>
          31st Januray, 2018
        </h5>
        <h5>
          1 min
        </h5>
      </div>
      <div
        class="cardTitle"
      >
        <h3>
          How painting outdoors can transform your 
        </h3>
      </div>
      <div
        class="cardDescription"
      >
        <h5>
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </h5>
      </div>
      <hr
        id="cardLine"
      />
      <div
        class="cardReaction"
      >
        <div
          class="clapCount"
        >
          <img
            class="clap-button"
            src="clapping.svg"
          />
          <h5
            class="clap"
          >
            1
          </h5>
        </div>
        <img
          alt="redHeart"
          class="like"
          src="heart-red.svg"
        />
      </div>
    </div>
    <div
      class="cardMain"
    >
      <div
        class="cardImage"
      >
        <img
          src="young-painters.png"
        />
      </div>
      <div
        class="cardTime"
      >
        <h5>
          20th March, 2018
        </h5>
        <h5>
          2 mins
        </h5>
      </div>
      <div
        class="cardTitle"
      >
        <h3>
          10 young painters you need to know
        </h3>
      </div>
      <div
        class="cardDescription"
      >
        <h5>
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </h5>
      </div>
      <hr
        id="cardLine"
      />
      <div
        class="cardReaction"
      >
        <div
          class="clapCount"
        >
          <img
            class="clap-button"
            src="clapping.svg"
          />
          <h5
            class="clap"
          >
            15
          </h5>
        </div>
        <img
          alt="blackHeart"
          class="like"
          src="heart-black.svg"
        />
      </div>
    </div>
    <div
      class="cardMain"
    >
      <div
        class="cardImage"
      >
        <img
          src="perfect-art.png"
        />
      </div>
      <div
        class="cardTime"
      >
        <h5>
          22nd June, 2018
        </h5>
        <h5>
          5 mins
        </h5>
      </div>
      <div
        class="cardTitle"
      >
        <h3>
          Tips for creating the perfect art
        </h3>
      </div>
      <div
        class="cardDescription"
      >
        <h5>
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </h5>
      </div>
      <hr
        id="cardLine"
      />
      <div
        class="cardReaction"
      >
        <div
          class="clapCount"
        >
          <img
            class="clap-button"
            src="clapping.svg"
          />
          <h5
            class="clap"
          >
            11
          </h5>
        </div>
        <img
          alt="blackHeart"
          class="like"
          src="heart-black.svg"
        />
      </div>
    </div>
    <div
      class="cardMain"
    >
      <div
        class="cardImage"
      >
        <img
          src="inspired.png"
        />
      </div>
      <div
        class="cardTime"
      >
        <h5>
          4th Feburary, 2019
        </h5>
        <h5>
          9 mins
        </h5>
      </div>
      <div
        class="cardTitle"
      >
        <h3>
          Stay inspired: top artists blog ...
        </h3>
      </div>
      <div
        class="cardDescription"
      >
        <h5>
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </h5>
      </div>
      <hr
        id="cardLine"
      />
      <div
        class="cardReaction"
      >
        <div
          class="clapCount"
        >
          <img
            class="clap-button"
            src="clapping.svg"
          />
          <h5
            class="clap"
          >
            45
          </h5>
        </div>
        <img
          alt="redHeart"
          class="like"
          src="heart-red.svg"
        />
      </div>
    </div>
    <div
      class="cardMain"
    >
      <div
        class="cardImage"
      >
        <img
          src="reason.png"
        />
      </div>
      <div
        class="cardTime"
      >
        <h5>
          10th Januray, 2019
        </h5>
        <h5>
          2 mins
        </h5>
      </div>
      <div
        class="cardTitle"
      >
        <h3>
          5 reasons why everyone should try an...
        </h3>
      </div>
      <div
        class="cardDescription"
      >
        <h5>
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </h5>
      </div>
      <hr
        id="cardLine"
      />
      <div
        class="cardReaction"
      >
        <div
          class="clapCount"
        >
          <img
            class="clap-button"
            src="clapping.svg"
          />
          <h5
            class="clap"
          >
            11
          </h5>
        </div>
        <img
          alt="blackHeart"
          class="like"
          src="heart-black.svg"
        />
      </div>
    </div>
    <div
      class="cardMain"
    >
      <div
        class="cardImage"
      >
        <img
          src="inspired.png"
        />
      </div>
      <div
        class="cardTime"
      >
        <h5>
          4th Feburary, 2019
        </h5>
        <h5>
          9 mins
        </h5>
      </div>
      <div
        class="cardTitle"
      >
        <h3>
          Stay inspired: top artists blog ...
        </h3>
      </div>
      <div
        class="cardDescription"
      >
        <h5>
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </h5>
      </div>
      <hr
        id="cardLine"
      />
      <div
        class="cardReaction"
      >
        <div
          class="clapCount"
        >
          <img
            class="clap-button"
            src="clapping.svg"
          />
          <h5
            class="clap"
          >
            45
          </h5>
        </div>
        <img
          alt="redHeart"
          class="like"
          src="heart-red.svg"
        />
      </div>
    </div>
    <div
      class="cardMain"
    >
      <div
        class="cardImage"
      >
        <img
          src="reason.png"
        />
      </div>
      <div
        class="cardTime"
      >
        <h5>
          10th Januray, 2019
        </h5>
        <h5>
          2 mins
        </h5>
      </div>
      <div
        class="cardTitle"
      >
        <h3>
          5 reasons why everyone should try an...
        </h3>
      </div>
      <div
        class="cardDescription"
      >
        <h5>
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </h5>
      </div>
      <hr
        id="cardLine"
      />
      <div
        class="cardReaction"
      >
        <div
          class="clapCount"
        >
          <img
            class="clap-button"
            src="clapping.svg"
          />
          <h5
            class="clap"
          >
            11
          </h5>
        </div>
        <img
          alt="blackHeart"
          class="like"
          src="heart-black.svg"
        />
      </div>
    </div>
    <div
      class="cardMain"
    >
      <div
        class="cardImage"
      >
        <img
          src="inspired.png"
        />
      </div>
      <div
        class="cardTime"
      >
        <h5>
          4th Feburary, 2019
        </h5>
        <h5>
          9 mins
        </h5>
      </div>
      <div
        class="cardTitle"
      >
        <h3>
          Stay inspired: top artists blog ...
        </h3>
      </div>
      <div
        class="cardDescription"
      >
        <h5>
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </h5>
      </div>
      <hr
        id="cardLine"
      />
      <div
        class="cardReaction"
      >
        <div
          class="clapCount"
        >
          <img
            class="clap-button"
            src="clapping.svg"
          />
          <h5
            class="clap"
          >
            45
          </h5>
        </div>
        <img
          alt="redHeart"
          class="like"
          src="heart-red.svg"
        />
      </div>
    </div>
    <div
      class="cardMain"
    >
      <div
        class="cardImage"
      >
        <img
          src="reason.png"
        />
      </div>
      <div
        class="cardTime"
      >
        <h5>
          10th Januray, 2019
        </h5>
        <h5>
          2 mins
        </h5>
      </div>
      <div
        class="cardTitle"
      >
        <h3>
          5 reasons why everyone should try an...
        </h3>
      </div>
      <div
        class="cardDescription"
      >
        <h5>
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </h5>
      </div>
      <hr
        id="cardLine"
      />
      <div
        class="cardReaction"
      >
        <div
          class="clapCount"
        >
          <img
            class="clap-button"
            src="clapping.svg"
          />
          <h5
            class="clap"
          >
            11
          </h5>
        </div>
        <img
          alt="blackHeart"
          class="like"
          src="heart-black.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
